src/overlays/actors/ovl_Obj_Driftice/z_obj_driftice.c

```diff
@@ -76,200 +87,257 @@ static InitChainEntry sInitChain[] = {
-    ObjDrifticeStruct3* temp_s0;
-    f32 temp_f22 = 0.0f;
-    s16 phi_s0;
+/*
+Updates dynamics of platform precession based on player interaction and populates tiltMagnitude and precessionAngle.
+    Value stored in tiltMagnitude approaches zero when player is not on platform.
+
+Magnitude is calculated based on two contributions.
+- tiltMagnitude1 = the combined output of two oscillation states * impactFromPlayer
+- tiltMagnitude2 is based on player position and contains a feedback loop where targetTiltMagnitude2 and
+    dynamics->tiltMagnitude2RateOfChange influence each other.
+*/
+void ObjDriftice_PlayerRelatedPrecessionPhysicsUpdate(ObjDrifticePlayerRelatedPrecessionDynamics* dynamics,
+                                                      ObjDriftice* this, s16* tiltMagnitude, s16* precessionAngle) {
+    ObjDrifticeOscillationState* oscillation;
+    f32 tiltMagnitude1 = 0.0f;
+    s16 targetTiltMagnitude2;
     s32 i;
 
-    if (this->unk_248 > 0) {
-        f32 temp_f20 = this->dyna.actor.xzDistToPlayer * this->unk_240 * 0.0007075472f;
-
-        if (temp_f20 > 1.0f) {
-            temp_f20 = 1.0f;
-        } else if (temp_f20 < 0.01f) {
-            temp_f20 = 0.01f;
+    if (this->timerPlayerOnTop > 0) {
+        f32 xzDistToPlayerNormalised = this->dyna.actor.xzDistToPlayer * this->inverseScale * 0.0007075472f; // 3 / 4240
+        if (xzDistToPlayerNormalised > 1.0f) {
+            xzDistToPlayerNormalised = 1.0f;
+        } else if (xzDistToPlayerNormalised < 0.01f) {
+            xzDistToPlayerNormalised = 0.01f;
         }
-
-        Math_StepToS(&arg0->unk_02, 1200.0f * temp_f20, 0x64);
-        phi_s0 = 2500.0f * temp_f20;
-        Math_StepToS(&arg0->unk_06, Math_CosS(arg0->unk_04 * 6.5536f) * (120.0f * temp_f20), 0x28);
-        Math_StepToF(&arg0->unk_08, 1.0f, 0.02f);
+        Math_StepToS(&dynamics->precessionAngleRateOfChange, 1200.0f * xzDistToPlayerNormalised, 100);
+        targetTiltMagnitude2 = 2500.0f * xzDistToPlayerNormalised;
+        // target for tiltMagnitudeRateOfChange will be:
+        // - higher if the platform is not already tilted much, or if player is far from the center
+        // - lower if the platform is already tilted, or if player is nearer to the center
+        Math_StepToS(&dynamics->tiltMagnitude2RateOfChange,
+                     Math_CosS(dynamics->tiltMagnitude2 * 6.5536f) * (120.0f * xzDistToPlayerNormalised), 40);
+        Math_StepToF(&dynamics->impactFromPlayer, 1.0f, 0.02f);
     } else {
-        Math_StepToS(&arg0->unk_02, 0, 0x96);
-        phi_s0 = 0;
-        Math_StepToS(&arg0->unk_06, 20, 7);
-        Math_StepToF(&arg0->unk_08, 0.0f, 0.02f);
+        Math_StepToS(&dynamics->precessionAngleRateOfChange, 0, 150);
+        targetTiltMagnitude2 = 0; // Tilt magnitude will decrease to zero over time
+        Math_StepToS(&dynamics->tiltMagnitude2RateOfChange, 20,
+                     7); // Tilt magnitude rate of change will quickly reach 20
```

Comment:
```suggestion
        // Tilt magnitude will decrease to zero over time
        targetTiltMagnitude2 = 0;
        
        // Tilt magnitude rate of change will quickly reach 20
        Math_StepToS(&dynamics->tiltMagnitude2RateOfChange, 20, 7); 
```
The comments going above will help the formatting here

Reply:
If someone knows what Math_StepToS does, I think the two "Tilt magnitude ..." comments can be removed. StepTo functions are quite common.